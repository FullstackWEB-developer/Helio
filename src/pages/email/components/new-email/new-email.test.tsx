import i18n from '../../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import NewEmail from './new-email';
import Router from "react-router-dom";
import EmailProvider from '@pages/email/context/email-context';
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("New Email tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders new-email correctly", async () => {
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' })

        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <EmailProvider>
                <NewEmail/>
            </EmailProvider>            
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders new-email with contact", async () => {
        const mockLocation = {
            pathname: '/new',
            hash: '',
            search: '',
            state: {
                contact: {

                }
            }
          }
        jest.spyOn(Router, 'useLocation').mockReturnValue(mockLocation)

        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <EmailProvider>
                <NewEmail/>
            </EmailProvider>            
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})
