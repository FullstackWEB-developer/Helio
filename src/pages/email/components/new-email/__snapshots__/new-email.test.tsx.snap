// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Email tests renders new-email correctly 1`] = `
<DocumentFragment>
  <div
    class="w-full h-full overflow-y-auto"
  >
    <div
      class="new-email-title-wrapper h-16 w-full flex items-center px-6 h6 justify-between"
    >
      <div
        class="w-1/2 new-email-title"
      >
        email.new_email.title
      </div>
      <div
        class="flex justify-end w-1/2 cursor-pointer"
      >
        <div
          class=""
        >
          <svg
            class="icon-medium "
            data-testid="go-back"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="white-icon"
              d="M16,1.409,14.591,0,8,6.566,1.409,0,0,1.409,6.566,8,0,14.591,1.409,16,8,9.434,14.591,16,16,14.591,9.434,8Z"
              transform="translate(4 4)"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row items-center w-full px-4 border-b"
    >
      <div
        class="pr-1 body2"
      >
        email.new_email.to
      </div>
      <div
        class="w-full"
      >
        <div
          class="h-16 w-full"
        >
          <div
            class="flex flex-row h-full"
          >
            <div
              class="relative h-10 w-full search-input-field flex items-center h-16"
            >
              <input
                autocomplete="off"
                class="py-2.5 h-full w-full search-input-field body2 border-none border-b"
                data-testid="search"
                placeholder="email.new_email.search_placeholder"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`New Email tests renders new-email with contact 1`] = `
<DocumentFragment>
  <div
    class="w-full h-full overflow-y-auto"
  >
    <div
      class="new-email-title-wrapper h-16 w-full flex items-center px-6 h6 justify-between"
    >
      <div
        class="w-1/2 new-email-title"
      >
        email.new_email.title
      </div>
      <div
        class="flex justify-end w-1/2 cursor-pointer"
      >
        <div
          class=""
        >
          <svg
            class="icon-medium "
            data-testid="go-back"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="white-icon"
              d="M16,1.409,14.591,0,8,6.566,1.409,0,0,1.409,6.566,8,0,14.591,1.409,16,8,9.434,14.591,16,16,14.591,9.434,8Z"
              transform="translate(4 4)"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row items-center w-full px-4 border-b"
    >
      <div
        class="pr-1 body2"
      >
        email.new_email.to
      </div>
      <div
        class="w-full"
      >
        <div
          class="h-16 w-full"
        >
          <div
            class="flex flex-row h-full"
          >
            <div
              class="relative h-10 w-full search-input-field flex items-center h-16"
            >
              <input
                autocomplete="off"
                class="py-2.5 h-full w-full search-input-field body2 border-none border-b"
                data-testid="search"
                placeholder="email.new_email.search_placeholder"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
